cost/forecasting/moving_average_forecast/moving_average_forecast.pt

```diff
@@ -0,0 +1,379 @@
+  var results = [];
+
+  var cost_metric = {
+    "Unamortized Unblended":"cost_nonamortized_unblended_adj",
+    "Amortized Unblended":"cost_amortized_unblended_adj",
+    "Unamortized Blended": "cost_nonamortized_blended_adj",
+    "Amortized Blended":"cost_amortized_blended_adj"
+  }
+
+  var mapped_costs = _.groupBy(ds_costs, function (value) {
+    return value.start_date;
+  });
+  var unsorted_results = _.map(mapped_costs, function (group) {
+    var counter = 0;
+    for (var i = 0; i < group.length; i++) {
+      var item = group[i];
+      if (item[cost_metric[param_cost_metric]] == 0) {
+        counter = counter + 1;
+      } else {
+        break;
+      }
+    }
+
+    var arr_new_sum = _.pluck(group, cost_metric[param_cost_metric]);
+
+    var summed = _.reduce(arr_new_sum, function (memo, num) { return memo + num; }, 0);
+    return {
+      start_date: group[0].start_date,
+      cost: summed,
+    };
+  });
+  var costs = [];
+  _.each(unsorted_results, function (item) {
+    costs.push({
+      start_date: item.start_date,
+      cost: item.cost,
+    });
+  });
+
+  var sorted_costs = _.sortBy(costs, function (o) { var dt = new Date(o.start_date); return -dt; });
+  console.log(sorted_costs);
+  results.push({
+    costs: sorted_costs,
+  });
+EOS
+end
+
+script "js_forecast", type: "javascript" do
+  parameters "ds_make_data_package", "param_average_months"
+  result "report"
+  code <<-EOS
+  var forecasted_costs = [];
+
+  // https://blog.oliverjumpertz.dev/the-moving-average-simple-and-exponential-theory-math-and-implementation-in-javascript
+  function simpleMovingAverage(prices, window) {
+    if (!prices || prices.length < window) {
+      return [];
+    }
+
+    var index = window - 1;
+    var length = prices.length + 1;
+
+    var simpleMovingAverages = [];
+    var counter = 0;
+    while (++index < length) {
+      var windowSlice = prices.slice(index - window, index);
+      var sum = _.reduce(windowSlice, function (prev, curr) {return  prev + curr; });
+      simpleMovingAverages.push( sum / window );
+      counter = counter + 1;
+    }
+
+    return simpleMovingAverages;
+  }
+
+
+  var costs = ds_make_data_package[0].costs;
+  console.log(costs);
+
+  var arr_costs = _.pluck(costs, 'cost');
+
+  forecasted_costs.push({ name: param_average_months + ' month average', average: simpleMovingAverage(arr_costs, param_average_months).reverse() });
+  size_difference = arr_costs.length - forecasted_costs[0]['average'].length
+  _.each(_.range(1, size_difference), function(item) {
+    forecasted_costs[0]['average'].push(forecasted_costs[0]['average'][forecasted_costs[0]['average'].length - 1])
+  })
+  var XAxisArray = _.range(1, arr_costs.length).reverse();
+  var chartData = 'chd=a:' + forecasted_costs[0]['average'].join(',') + '|' + arr_costs.reverse().join(',');
+  var chartXAxis = 'chxl=0:|' + XAxisArray.join('|');
+  forecasted_costs[0]['average'] = forecasted_costs[0]['average'].reverse()
+  var report = {
```

Comment:
Need to include encodeURI on each chart parameter for image-chart's API function. 320-333